jax/_src/numpy/ufuncs.py

```diff
@@ -587,19 +587,127 @@ def tan(x: ArrayLike, /) -> Array:
   """
   return lax.tan(*promote_args_inexact('tan', x))
 
-@implements(np.arcsin, module='numpy')
+
 @partial(jit, inline=True)
 def arcsin(x: ArrayLike, /) -> Array:
+  r"""Compute element-wise inverse of trigonometric sine of input.
+
+  JAX implementation of :obj:`numpy.arcsin`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array containing the inverse trigonometric sine of each element of ``x``
+    in radians in the range ``[-pi/2, pi/2]``, promoting to inexact dtype.
+
+  Note:
+    - ``jnp.arcsin`` returns ``nan`` when ``x`` is real-valued and not in the closed
+      interval ``[-1, 1]``.
+    - ``jnp.arcsin`` follow the branch cut convention used by the associated NumPy
```

Comment:
Rather than "the associated numpy function", I'd explicitly say that it ```follows the branch cut convention of :func:`numpy.arcsin` ```.

Same with the other functions below, linking to their associated numpy function.

Reply:
Thanks! Updated as suggested.